agent/agent.go

```diff
@@ -1267,15 +1065,15 @@ func (a *Agent) shouldKill(tc *taskContext, ignoreTaskGroupCheck bool) bool {
 		return false
 	}
 	// kill if the task is not in a task group
-	if tc.taskConfig.TaskGroup != nil && tc.taskConfig.TaskGroup.Name == "" {
+	if tc.taskConfig.TaskGroup == nil {
 		return true
 	}
 	// kill if ignoreTaskGroupCheck is true
 	if ignoreTaskGroupCheck {
 		return true
 	}
 	// do not kill if share_processes is set
-	if tc.taskConfig.TaskGroup.Name != "" && tc.taskConfig.TaskGroup.ShareProcs {
+	if tc.taskConfig.TaskGroup != nil && tc.taskConfig.TaskGroup.ShareProcs {
```

Comment:
This can just be `if tc.taskConfig.TaskGroup.ShareProcs` since an earlier block guarantees the task group is not nil

Reply:
We actually can just remove this if statement and call `return !tc.taskConfig.TaskGroup.ShareProcs`

Reply:
Done.